src/main/java/build/buildfarm/worker/shard/Worker.java

```diff
@@ -311,30 +320,119 @@ private ExecFileSystem createFuseExecFileSystem(
         storage);
   }
 
-  private @Nullable UserPrincipal getOwner(FileSystem fileSystem) throws ConfigurationException {
+  private @Nullable UserPrincipal getOwner(String name, FileSystem fileSystem)
+      throws ConfigurationException {
     try {
-      return getUser(configs.getWorker().getExecOwner(), fileSystem);
+      return getUser(name, fileSystem);
     } catch (IOException e) {
       ConfigurationException configException =
-          new ConfigurationException("Could not locate exec_owner");
+          new ConfigurationException("Could not locate " + name);
       configException.initCause(e);
       throw configException;
     }
   }
 
+  private static final class Unqueue<T> extends AbstractCollection<T> implements Queue<T> {
```

Comment:
todo: a pinch of javadoc? the class name doesn't immediately tell me the purpose..

Reply:
Can add. If you have a good idea to name a thing like a queue that only puts out the same object, I'm all ears. Hose? Fountain? CloneGun?

Reply:
Settled on a Dispenser and added javadoc